syncer/syncer.go

```diff
@@ -1098,9 +1011,18 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 	)
 	s.tctx.L().Info("replicate binlog from checkpoint", zap.Stringer("checkpoint", lastPos))
 
-	s.streamer, err = s.streamerProducer.generateStreamer(lastPos)
-	if err != nil {
-		return err
+	if s.streamerController == nil {
+		s.Lock()
```

Comment:
This is a bit hacky. How about creating an idle `StreamerController` in `NewSyncer`?

Reply:
refine in 3910ffc, create `StreamerController` in `Init`, and Start it in `run`